nixos/modules/services/web-apps/netbox.nix

```diff
@@ -0,0 +1,227 @@
+
+      ${cfg.extraConfig}
+    '';
+  };
+  pkg = (pkgs.netbox.overrideAttrs (old: {
+    installPhase = old.installPhase + ''
+      ln -s ${configFile} $out/opt/netbox/netbox/netbox/configuration.py
+    '' + optionalString cfg.enableLdap ''
+      ln -s ${ldapConfigPath} $out/opt/netbox/netbox/netbox/ldap_config.py
+    '';
+  })).override {
+    plugins = ps: ((cfg.plugins ps)
+      ++ optional cfg.enableLdap [ ps.django-auth-ldap ]);
+  };
+  netboxManageScript = with pkgs; (writeScriptBin "netbox-manage" ''
+    #!${stdenv.shell}
+    export PYTHONPATH=${pkg.pythonPath}
+    sudo -u netbox ${pkg}/bin/netbox "$@"
+  '');
+  defaultServiceConfig = {
+    WorkingDirectory = "${cfg.dataDir}";
+    User = "netbox";
+    Group = "netbox";
+    StateDirectory = "netbox";
+    StateDirectoryMode = "0750";
+    Restart = "on-failure";
+  };
+
+in {
+  options.services.netbox = {
+    enable = mkEnableOption "Netbox";
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "[::1]";
+      description = ''
+        Address the server will listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8001;
+      description = ''
+        Port the server will listen on.
+      '';
+    };
+
+    plugins = mkOption {
+      type = types.functionTo (types.listOf types.package);
+      default = _: [];
+      defaultText = literalExpression ''
+        python3Packages: with python3Packages; [];
+      '';
+      description = ''
+        Python-Packages to install as Plugins.
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/netbox";
+      description = ''
+        Storage path of netbox.
+      '';
+    };
+
+    secretKeyFile = mkOption {
+      type = types.path;
+      description = ''
+        A file containing the secret key.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Additional lines of configuration appended to the <literal>configuration.py</literal>.
+        See https://netbox.readthedocs.io/en/stable/configuration/optional-settings/ for more details.
+      '';
+    };
+
+    enableLdap = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable LDAP-Authentication for Netbox.
+        Configuration for this is done through <literal>ldapConfig</literal>.
+      '';
+    };
+
+    ldapConfigPath = mkOption {
+      type = types.path;
+      default = "";
+      description = ''
+        Path to the Configuration-File for LDAP-Authentification, will be loaded as <literal>ldap_config.py</literal>.
+        See https://netbox.readthedocs.io/en/stable/installation/6-ldap/#configuration for more details.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.redis.servers.netbox.enable = true;
+
+    services.postgresql = {
+      enable = true;
```

Comment:
I'm torn on modules implicitly enabling a database service and I'm not exactly sure what our stance on this is. Maybe @aanderse knows?

Reply:
This wouldn't be the first time we do it. I think the same comment you made about `redis` assumptions applies here.

This PR is the first time someone can run a NixOS provided version of this software so I think we can leave the hard coded database configuration as is. In the future I would assume someone will add a set of `database` options, including `database.createLocally` (a common pattern in NoxOS modules), so this can be made configurable. **Eventually** someone will file an issue requesting this feature, but not today.

Normally I would offer to help and could add the options for you, but I'm a bit overloaded at the moment.